build.gradle.kts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2017-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import kotlinx.validation.*
+import org.jetbrains.dokka.gradle.*
+
+plugins {
+    base
+    alias(libs.plugins.knit)
+    id("org.jetbrains.kotlinx.binary-compatibility-validator")
```

Comment:
Because of `buildSrc` have `id("org.jetbrains.kotlinx.binary-compatibility-validator")` thus, BCV falls into the plug-in's `classpath`

Reply:
> There appear to be some issues in using buildSrc with dependencies (instead includeBuild works better).

This is the expected behavior. From [docs](https://docs.gradle.org/current/userguide/organizing_gradle_projects.html#sec:build_sources):  `Upon discovery of the directory, Gradle automatically compiles and tests this code and puts it in the classpath of your build script.`

Therefore, if `implementation` dependency is specified in buildSrc, then this dependency is also placed in the classpath, as well as the classes declared in the buildSrc itself. Thus, they become available from any build script.

Reply:
> Ah, so you've put all plugins as dependencies in buildSrc. Is it to conveniently use version catalogs or for something else?

Gradle requires each plugin to be placed in the build classpath only once.

In order to use the plugin as inside `buildSrc`, we will in any case have to use it as a dependency. Thus, he gets into plugins classpath. 
And if we specify the plugin version in the root project (this is equivalent to including plugin to the classpath for the current project and all subprojects ), we will get an error 
`The request for this plugin could not be satisfied because the plugin is already on the classpath with an unknown version, so compatibility cannot be checked.`

Reply:
It seems that this approach complicates understanding, so I'll try to replace the dependence in buildSrc with `compileOnly` (in this case, the dependence from buildSrc will not get into classpath)